src/index.ts

```diff
@@ -661,6 +690,48 @@ function makeSourceMap(
   };
 }
 
+/**
+ * This method maps the newly generated @param{sourceMap} to the input source map.
+ * This is required when ts-loader is not the first loader in the Webpack loader chain.
+ * If the input source map is not set, then the newly generated source map is returned.
+ */
+function mapToInputSourceMap(
+  sourceMap: RawSourceMap,
+  loaderContext: webpack.LoaderContext<LoaderOptions>,
+  inputSourceMap: Record<string, any>
+): Promise<RawSourceMap> {
+  return new Promise<RawSourceMap>((resolve, reject) => {
+    const inMap = inputSourceMap as RawSourceMap;
```

Comment:
Rather than casting the incoming source map as a `RawSourceMap` and mutating it, could we create a `RawSourceMap` based upon the `inputSourceMap`?

Reply:
Looking at the type definition, these seem like quite different types?

https://github.com/mozilla/source-map/blob/60adcb064bf033702d954d6d3f9bc3635dcb744b/source-map.d.ts#L15-L21

Reply:
Webpack declares this type, which is the same up to some optionals: https://github.com/webpack/webpack/blob/1f13ff9fe587e094df59d660b4611b1bd19aed4c/types.d.ts#L11978C2-L11986C2  

Unfortunately, they don't export it, which is why I was using the `Record<string, any>` type. I could potentially re-declare the interface inside ts-loader but I think simply casting the type should be sufficient in this case.

How would you do this?

Reply:
Let's ignore the casting issue for now - how would you feel about doing this?

> Rather than casting the incoming source map as a `RawSourceMap` and mutating it, could we create a `RawSourceMap` based upon the `inputSourceMap`?

Reply:
I've changed that part so that I don't modify the original object.

Reply:
Thanks - let's see if the existing tests pass